File: shared/schemas/calendar.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class CalendarEvent(BaseModel):
    calendar_event_id: UUID
    ticket_id: UUID
    start_time: datetime
    end_time: datetime
    duration: float
    status: str


class ScheduleBlock(BaseModel):
    day: str
    start_time: datetime
    end_time: datetime
    duration: float
    calendar_event_id: UUID | None = None

File: shared/types/calendar.ts
export interface CalendarEvent {
  calendar_event_id: string;
  ticket_id: string;
  start_time: string;
  end_time: string;
  duration: number;
  status: string;
}

export interface ScheduleBlock {
  day: string;
  start_time: string;
  end_time: string;
  duration: number;
  calendar_event_id: string | null;
}
